server/src/main/java/org/elasticsearch/index/mapper/DynamicTemplate.java

```diff
@@ -271,7 +275,9 @@ static DynamicTemplate parse(String name, Map<String, Object> conf) throws Mappe
             } else if ("path_unmatch".equals(propName)) {
                 addEntriesToPatternList(pathUnmatch, propName, entry);
             } else if ("match_mapping_type".equals(propName)) {
-                matchMappingType = entry.getValue().toString();
+                matchMappingType = entry.getValue();
```

Comment:
Can we use `addEntriesToPatternList` here?

Reply:
Yes, done. Also now using `matchPatternsAreDefined` for the implicit wildcard, and made it a varargs method.